visual-inspection/inspection-frontend/src/App.vue

```diff
@@ -135,7 +135,7 @@ export default {
         paragraphs: [this.$t('info2paragraph1'), this.$t('info2paragraph2'), this.$t('info2paragraph3')]
       }],
       backendUrl: process.env.VUE_APP_BACKEND_URL,
-      img: require('./assets/table.jpg')
+      img: require('./assets/desk.jpg') // other use case image: change .jpg-part to any other file from assets/
```

Comment:
Das sollte "von außen" konfigurabel sein, nicht durch Änderungen im Code bestimmt.

Schau dir mal an wie in der Zeile obendrüber die Backend-URL definiert ist und dann in diese Datei: https://github.com/XAI-Demonstrator/xai-demonstrator/blob/master/visual-inspection/inspection-frontend/.env.prod-deployment

Diese Umgebungen werden bei "npm build" mitgegeben, z.B. so wie hier: https://github.com/XAI-Demonstrator/xai-demonstrator/blob/master/landing-page/Dockerfile#L6

Wenn das für dich Sinn ergibt, bau das genau so nach, ansonsten schauen wir uns das gerne einmal zusammen an!

Reply:
Meint "von außen", dass auch die Schüler die Möglichkeit haben diese Änderung herbei zu führen?

Reply:
Oder reden wir von einer Art configurationsdatei?

Reply:
Die Schüler sollen das nicht ändern, aber wir wollen das Frontend in verschiedenen Konfigurationen bauen können, ohne dafür den Code zu ändern.

Die Backend-URL ist ein gutes Beispiel dafür: Die ist anders, je nachdem, ob das Frontend direkt mit im Docker-Container ist, in einem separaten Landing-Page-Container ausgeliefert wird (so wie im Test-Deployment) oder woanders gehostet wird (so wie beim Prod-Deployment, da werden die Frontends über Firebase Hosting ausgespielt).

Wir definieren die URL (und zukünftig dann auch das Bild) über die `.env`-Dateien (z.B. https://github.com/XAI-Demonstrator/xai-demonstrator/blob/master/visual-inspection/inspection-frontend/.env.prod-deployment). Ich kann mir schon vorstellen, dass wir das mittelfristig so lösen, dass das Frontend sich beim Backend (oder woanders, z.B. bei Firebase) erkundigt, welches Bild und welche Konfiguration gezeigt werden soll. Das sollten wir aber nur einbauen, wenn wir das wirklich mal brauchen sollten. 

Langer Rede kurzer Sinn: Du kannst hier einfach `require('./assets/' + process.env.IMAGE_FILE + ')` eintragen und dann die bestehenden .env-Dateien anpassen (die sollten alle auf table.jpg lauten). Für Digital Education machst du eine neue `.env.digital-education` wo stattdessen `desk.jpg` als `IMAGE_FILE` drinsteht und dann kannst du mit `npm run serve -- --mode digital-education` diese Version starten.

Reply:
Meintest du "require('./assets/' + process.env.IMAGE_FILE) ?